tools/dep-collector/README.md

```diff
@@ -1,5 +1,9 @@
 # dep-collector
 
+> Deprecation notice: the CI/CD flow has switched the license tool to [go-licenses](https://github.com/google/go-licenses),
+> and deprecated `dep-collector`. We need to keep it for now as the old release branches are still
+> using it. It can be safely deleted after we only support branches >= release-0.15.
```

Comment:
<!--mdfmt-bot-->
Format markdown:
```suggestion
> Deprecation notice: the CI/CD flow has switched the license tool to
> [go-licenses](https://github.com/google/go-licenses), and deprecated
> `dep-collector`. We need to keep it for now as the old release branches are
> still using it. It can be safely deleted after we only support branches >=
> release-0.15.
```